tests/unit/SpecialResolution.spec.ts

```diff
@@ -0,0 +1,367 @@
+    'FnZS11c2VycyIsInF1ZXJ5LXJlYWxtcyIsInZpZXctYXV0aG9yaXphdGlvbiIsInF1ZXJ5LWNsaWVudHMiLCJxdWVyeS11c2VycyIsIm1hbmF' +
+    'nZS1ldmVudHMiLCJtYW5hZ2UtcmVhbG0iLCJ2aWV3LWV2ZW50cyIsInZpZXctdXNlcnMiLCJ2aWV3LWNsaWVudHMiLCJtYW5hZ2UtYXV0aG9y' +
+    'aXphdGlvbiIsIm1hbmFnZS1jbGllbnRzIiwicXVlcnktZ3JvdXBzIl19LCJicm9rZXIiOnsicm9sZXMiOlsicmVhZC10b2tlbiJdfSwiYWNjb' +
+    '3VudCI6eyJyb2xlcyI6WyJtYW5hZ2UtYWNjb3VudCIsIm1hbmFnZS1hY2NvdW50LWxpbmtzIiwidmlldy1wcm9maWxlIl19fSwic2NvcGUiOi' +
+    'JvcGVuaWQiLCJmaXJzdG5hbWUiOiJTdW1lc2giLCJyb2xlcyI6WyJ2aWV3IiwiZWRpdCIsIm9mZmxpbmVfYWNjZXNzIiwic3RhZmYiLCJ1bWF' +
+    'fYXV0aG9yaXphdGlvbiIsImJhc2ljIl0sIm5hbWUiOiJTdW1lc2ggS2FyaXlpbCIsInByZWZlcnJlZF91c2VybmFtZSI6InNrYXJpeWlsQGlk' +
+    'aXIiLCJlbWFpbCI6InN1bWVzaC5wLmthcml5aWxAZ292LmJjLmNhIiwibGFzdG5hbWUiOiJLYXJpeWlsIiwidXNlcm5hbWUiOiJza2FyaXlpb' +
+    'EBpZGlyIn0.MSPSakOnCUia4qd-fUpvP2PB3k977Eyhjxn-ykjadsUTEK4f2R3c8vozxaIIMH0-qUwduyQmdZCl3tQnXYQ9Ttf1PE9eMLS4sX' +
+    'JiIUlDmKZ2ow7GmmDabic8igHnEDYD6sI7OFYnCJhRdgVEHN-_4KUk2YsAVl5XUr6blJKMuYDPeMyNreGTXU7foE4AT-93FwlyTyFzQGddrDv' +
+    'c6kkQr7mgJNTtgg87DdYbVGbEtIetyVfvwEF0rU8JH2N-j36XIebo33FU3-gJ5Y5S69EHPqQ37R9H4d8WUrHO-4QzJQih3Yaea820XBplJeo0' +
+    'DO3hQoVtPD42j0p3aIy10cnW2g')
+  store.state.stateModel.tombstone.businessId = 'CP1234567'
+
+  beforeEach(async () => {
+    // mock the window.location.assign function
+    delete window.location
+    window.location = { assign: jest.fn() } as any
+
+    const get = sinon.stub(axios, 'get')
+
+    // GET payment fee for immediate alteration
+    get.withArgs('https://pay.api.url/fees/CP/SPRLN')
+      .returns(Promise.resolve({
+        data: { 'filingFees': 70.0,
+          'filingType': 'Special resolution',
+          'filingTypeCode': 'SPRLN',
+          'futureEffectiveFees': 0,
+          'priorityFees': 0,
+          'processingFees': 0,
+          'serviceFees': 0,
+          'tax': {
+            'gst': 0,
+            'pst': 0
+          },
+          'total': 70.0
+        }
+      }))
+
+    // GET payment fee for future alteration
+    get.withArgs('https://pay.api.url/fees/CP/SPRLN?futureEffective=true')
+      .returns(Promise.resolve({
+        data: { 'filingFees': 70.0,
+          'filingType': 'Special resolution',
+          'filingTypeCode': 'SPRLN',
+          'futureEffectiveFees': 0,
+          'priorityFees': 0,
+          'processingFees': 0,
+          'serviceFees': 0,
+          'tax': {
+            'gst': 0,
+            'pst': 0
+          },
+          'total': 70.0 }
+      }))
+
+    // GET business
+    get.withArgs('businesses/CP1234567')
+      .returns(Promise.resolve({
+        data: {
+          business: {
+            legalName: 'Mock Business Ltd.',
+            legalType: 'CP',
+            identifier: 'CP1234567'
+          }
+        }
+      }))
+
+    // GET business name translations
+    get.withArgs('businesses/CP1234567/aliases')
```

Comment:
Does special resolution need name translation? If not, please clean up.

Same for other mocks here.

Reply:
I think you still have other mocked API calls that you don't need.